src/diagnostics/MOM_harmonic_analysis.F90

```diff
@@ -60,13 +62,15 @@ module MOM_harmonic_analysis
 
 !> This subroutine sets static variables used by this module and initializes CS%list.
 !! THIS MUST BE CALLED AT THE END OF tidal_forcing_init.
-subroutine HA_init(Time, US, param_file, time_ref, nc, freq, phase0, const_name, CS)
+subroutine HA_init(Time, US, param_file, time_ref, nc, freq, phase0, const_name, tide_fn, tide_un, CS)
```

Comment:
Would it be more concise if `tidal_forcing_CS` is used as a single input, rather than its members?

Reply:
This is because `tidal_forcing_CS` is not transparent and `MOM_harmonic_analysis` cannot see what's inside. @Hallberg-NOAA commented on this in his [review](https://github.com/NOAA-GFDL/MOM6/pull/603#issuecomment-2071197780) of the initial PR for inline harmonic analysis.